simulator/examples/clock.rs

```diff
@@ -0,0 +1,182 @@
+        let start = Point::new(
+            CENTER + (angle.cos() * (CENTER as f32)) as i32,
+            CENTER + (angle.sin() * (CENTER as f32)) as i32,
+        );
+
+        // End point; start point offset by `tic_len` pixels towards the circle center
+        let end = start
+            - Point::new(
+                (angle.cos() * tic_len) as i32,
+                (angle.sin() * tic_len) as i32,
+            );
+
+        Line::new(start, end)
+            .stroke(Some(BinaryColor::On))
+            .into_iter()
+    });
+
+    // Create a single iterator of pixels, first iterating over the circle, then over the 12 lines
+    // generated
+    face.into_iter().chain(tics.flatten())
+}
+
+/// Draw the seconds hand given a seconds value (0 - 59)
+fn draw_seconds_hand(seconds: u32) -> impl Iterator<Item = Pixel<BinaryColor>> {
+    // Convert seconds into a position around the circle in radians
+    let seconds_radians = ((seconds as f32 / 60.0) * 2.0 * PI) + START;
+
+    let end = Point::new(
+        CENTER + (seconds_radians.cos() * (CENTER as f32)) as i32,
+        CENTER + (seconds_radians.sin() * (CENTER as f32)) as i32,
+    );
+
+    // Basic line hand
+    let hand = Line::new((CENTER, CENTER).into(), end).stroke(Some(BinaryColor::On));
+
+    // Offset from end of hand
+    let decoration_offset = Point::new(
+        (seconds_radians.cos() * (20.0)) as i32,
+        (seconds_radians.sin() * (20.0)) as i32,
+    );
+
+    // Add a fancy circle near the end of the hand
+    let decoration = Circle::new(end - decoration_offset, 5)
+        .fill(Some(BinaryColor::Off))
+        .stroke(Some(BinaryColor::On))
+        .stroke_width(1);
+
+    hand.into_iter().chain(decoration)
+}
+
+/// Draw the hour hand (0-11)
+fn draw_hour_hand(hour: u32) -> Line<BinaryColor> {
+    // Convert hour into a position around the circle in radians
+    let hour_radians = ((hour as f32 / 12.0) * 2.0 * PI) + START;
+
+    let hand_len = CENTER as f32 - 60.0;
+
+    let end = Point::new(
+        CENTER + (hour_radians.cos() * hand_len) as i32,
+        CENTER + (hour_radians.sin() * hand_len) as i32,
+    );
+
+    // Basic line hand
+    Line::new((CENTER, CENTER).into(), end).stroke(Some(BinaryColor::On))
+}
+
+/// Draw the minute hand (0-59)
+fn draw_minute_hand(minute: u32) -> Line<BinaryColor> {
+    // Convert minute into a position around the circle in radians
+    let minute_radians = ((minute as f32 / 60.0) * 2.0 * PI) + START;
+
+    let hand_len = CENTER as f32 - 30.0;
+
+    let end = Point::new(
+        CENTER + (minute_radians.cos() * hand_len) as i32,
+        CENTER + (minute_radians.sin() * hand_len) as i32,
+    );
+
+    // Basic line hand
+    Line::new((CENTER, CENTER).into(), end).stroke(Some(BinaryColor::On))
+}
+
+fn main() {
+    let mut display = DisplayBuilder::new()
+        .title("Clock")
+        .size(DISP_SIZE as usize, DISP_SIZE as usize)
+        .scale(2)
+        .build_rgb();
```

Comment:
Why do you use an RGB display?

Reply:
Oops, I wanted to see how colour conversions were handled. I've changed this to `build_binary()`.